From 1ec373d6c1cb2221ce7460d034d6fdad0cdf3369 Mon Sep 17 00:00:00 2001
From: leihuang <leihuang@bestechnic.com>
Date: Thu, 9 Sep 2021 11:53:55 +0800
Subject: [PATCH] feat: build libqrcodegen.a for liteos_m

Signed-off-by: leihuang <leihuang@bestechnic.com>
Change-Id: I47d2099d41e7acf314940e09d49f6242c6e673e7
---
 BUILD.gn | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index 73f6763..942f1eb 100755
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -19,13 +19,15 @@ config("qrcodegen_config") {
 }
 
 if (defined(ohos_lite)) {
-  shared_library("qrcodegen") {
+  lite_library("qrcodegen") {
+    if (ohos_kernel_type == "liteos_m") {
+      target_type = "static_library"
+    } else {
+      target_type = "shared_library"
+    }
     sources = [ "cpp/QrCode.cpp" ]
     include_dirs = [ "//third_party/qrcodegen/cpp" ]
-    cflags = [
-      "-Wall",
-      "-fno-exceptions",
-    ]
+    cflags = [ "-Wall" ]
     cflags_cc = cflags
     public_configs = [ ":libqrcodegen_config" ]
   }
-- 
2.17.1

